openerp/modules/registry.py

```diff
@@ -457,8 +457,7 @@ def check_registry_signaling(cls, db_name):
         changed = False
         if openerp.multi_process and db_name in cls.registries:
             registry = cls.get(db_name)
-            cr = registry.cursor()
-            try:
+            with registry.cursor() as cr:
```

Comment:
Should be `with(closing(registry.cursor()))` for identical behaviour, otherwise the transaction will be committed. Shouldn't have any impact though, possibly slightly less traffic (cursors are rollbacked when closed, so a regular `with cursor` will commit then rollback I think)